txp/Pos/Binary/Txp/Core.hs

```diff
@@ -18,11 +18,30 @@ import qualified Pos.Txp.Core.Types as T
 -- Core
 ----------------------------------------------------------------------------
 
+instance Bi T.UtxoTxIn where
+    encode utxIn =
+        encodeListLen 2 <>
+        encode (T.txInHash utxIn) <>
+        encode (T.txInIndex utxIn)
+    decode = do
+        enforceSize "UtxoTxIn" 2
+        T.UtxoTxIn <$> decode <*> decode
+
 instance Bi T.TxIn where
-  encode txIn = encodeListLen 2 <> encode (T.txInHash txIn) <> encode (T.txInIndex txIn)
-  decode = do
-    enforceSize "TxIn" 2
-    T.TxIn <$> decode <*> decode
+    encode (T.TxInUtxo utxoTxIn) =
+        encodeListLen 2 <>
+        encode (0 :: Word8) <>
+        encode (serialize' utxoTxIn)
```

Comment:
I don't like how it's done for `TxInWitness`, but it's subjective and it seems that majority likes it, so I don't mind. But there is a bug, note `deserialize'` in `decode @TxInWitness`.